File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        if(s.length() != t.length())
            return false;
        
        int length = s.length();
        Map<Character, Character> charMappedMap = new HashMap<>();

        for(int i = 0 ; i < length ; i++)
        {
            char original = s.charAt(i);
            char replacement = t.charAt(i);

            if(!charMappedMap.containsKey(original))
            {
                if(!charMappedMap.containsValue(replacement))
                    charMappedMap.put(original, replacement);
                else
                    return false;
            }
            else
            {
                char mappedChar = charMappedMap.get(original);

                if(mappedChar != replacement) 
                    return false;
            }
        }

        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    
    if(s.length != t.length)
        return false;
    
    let length = s.length;
    let charMappedMap = new Map();

    for(let i = 0 ; i < length ; i++)
    {
        let original = s.charAt(i);
        let replacement = t.charAt(i);

        if(!charMappedMap.has(original))
        {
            if(!checkValue(charMappedMap,replacement))
                charMappedMap.set(original, replacement);
            else
                return false;

        }
        else
        {
            let mappedChar = charMappedMap.get(original);

            if(mappedChar != replacement)
                return false;
        }
    }

    return true;
};

function checkValue(charMappedMap,replacement)
{
    for(let value of charMappedMap.values())
    {
        if(value == replacement)
            return true;
    }

    return false;
}